tests/x11/sshxterm.pm

```diff
@@ -16,19 +16,22 @@ sub run() {
     mouse_hide(1);
     x11_start_program("xterm");
     script_run("ssh -XC root\@localhost xterm");
-    type_string "yes\n";
-    wait_idle 6;
+    my $ret = check_screen([qw/ssh-xterm-host-key-authentication ssh-password-prompt/]);
+    die "ssh did not respond in time or needle(s) not found" unless $ret;
+    # if ssh asks for authentication of the key accept it
+    if ($ret->{needle}->has_tag('ssh-xterm-host-key-authentication')) {
+        type_string "yes\n";
+        assert_screen "ssh-password-prompt";
+    }
     type_string "$password\n";
-    sleep 2;
+    assert_screen "ssh-second-xterm";
     for (1 .. 13) { send_key "ret" }
     $self->set_standard_prompt();
     type_string "echo If you can see this text, ssh-X-forwarding  is working.\n";
-    sleep 2;
-    assert_screen 'test-sshxterm-1', 3;
+    assert_screen 'test-sshxterm-1';
+    # close both windows
     send_key "alt-f4";
-    sleep 1;
     send_key "alt-f4";
```

Comment:
yes, I was fearing the same. So maybe we actually need another assert_screen here. But basically we are already in the "cleanup" phase. Any other idea how we could handle this safely? And more important: How can I test this locally :-)